app/apptesting/test_helpers.go

```diff
@@ -77,7 +78,19 @@ func SetupTestingApp() (*app.App, app.GenesisState) {
 	params.SetAddressPrefixes()
 
 	newApp := app.New(log.NewNopLogger(), db, nil, true, map[int64]bool{}, app.DefaultNodeHome, 5, encCdc, EmptyAppOptions{})
-	return newApp, app.NewDefaultGenesisState(encCdc.Codec)
+
+	defaultGenesisState := app.NewDefaultGenesisState(encCdc.Codec)
+
+	// set EnableCreate to false
+	if evmGenesisStateJson, found := defaultGenesisState[evmtypes.ModuleName]; found {
+		// force disable Enable Create of x/evm
+		var evmGenesisState evmtypes.GenesisState
+		encCdc.Codec.MustUnmarshalJSON(evmGenesisStateJson, &evmGenesisState)
+		evmGenesisState.Params.EnableCreate = false
+		defaultGenesisState[evmtypes.ModuleName] = encCdc.Codec.MustMarshalJSON(&evmGenesisState)
+	}
```

Comment:
```suggestion

	defaultGenesisState := make(map[string]json.RawMessage)

	for _, b := range app.ModuleBasics {
		if b.Name() == evmtypes.ModuleName {
			evmGenesisState := evmtypes.DefaultGenesisState()
			// force disable Enable Create of x/evm
			evmGenesisState.Params.EnableCreate = false
			defaultGenesisState[b.Name()] = encCdc.Codec.MustMarshalJSON(evmGenesisState)
			continue
		}
		defaultGenesisState[b.Name()] = b.DefaultGenesis(encCdc.Codec)
	}
```

this could save us an extra unmarshalling and marshalling